### PR TITLE
Playwright_Properly check for the requested tabs

### PR DESCRIPTION
**Pull request by Rinchan20**
This PR is a fix for the findParticipantWithTab(...) method which is used to find a participants with a specific tab since tests are failing to find a participant as intended - so far, the changes made have fixed the following tests (based on local run):

- RGP Blood and RNA
- Multi tab
- Osteo adult and child enrollment (consent-assent)
- Osteo child enrollment (parental-consent)
- Editing Participant Information (for 3 studies: LMS, OS2, PanCan)
- Update Institution in Medical Request page
- LMS Test for checking SAMPLES_RECEIVED

Edit: PW tests failures are not just due to the aforementioned method - other tweaks are needed

(see this morning's PW tests failures here: https://app.circleci.com/pipelines/github/broadinstitute/ddp-angular/26780/workflows/573feb68-f9f4-47fe-8cdf-838b07a7ac83/jobs/100644/tests)

### CODE REVIEW

File: playwright-e2e/dsm/pages/participant-list-page.ts [hunk 1]
```diff
@@ -263,14 +264,34 @@ export default class ParticipantListPage extends DsmPageBase {
...
+        const participantData = responseJson.participants[index].participantData[0];
         firstName = responseJson.participants[index].esData.profile.firstName;
         if (participantData && (firstName === testParticipantFirstName)) {
           shortID = responseJson.participants[index].esData.profile.hruid;
           logInfo(`Found the RGP participant ${shortID} to have a proband tab`);
           break;
         }
       }
+
+      //Go to the next page if none of the participants in the current page are relevant
```
Comment (Rinchan20) [inline playwright-e2e/dsm/pages/participant-list-page.ts]: planning to move this to be a general method in a separate PR since it ended up being used twice

File: playwright-e2e/dsm/pages/participant-list-page.ts [hunk 2]
```diff
@@ -263,14 +264,34 @@ export default class ParticipantListPage extends DsmPageBase {
 
       if (rgpProbandTab === true) {
         //If the participant has participantData, this seems to mean there's a proband tab in the account
-        const participantData = responseJson.participants[index].participantData;
+        const participantData = responseJson.participants[index].participantData[0];
```
Comment (aweng98) [inline playwright-e2e/dsm/pages/participant-list-page.ts]: Any chance `participantData` is undefined? If so, `participantData[0]` will throw exception and it's not handled.

File: playwright-e2e/utils/test-utils.ts [hunk 3]
```diff
@@ -36,16 +38,25 @@ export async function waitForNoSpinner(page: Page, opts: { timeout?: number } =
...
-export async function waitForResponse(page: Page, { uri, status = 200, timeout }: WaitForResponse): Promise<Response> {
-  const response = await page.waitForResponse((response: Response) => response.url().includes(uri), { timeout });
+export async function waitForResponse(page: Page, { uri, status = 200, timeout, messageBody }: WaitForResponse): Promise<Response> {
+  let response: any;
+  if (messageBody) {
+    response = await page.waitForResponse((response: Response) => response.url().includes(uri) &&
+    messageBody.every(async message => (await response.text()).includes(message as string)), { timeout });
+  }
+
+  if (!messageBody) {
```
Comment (aweng98) [inline playwright-e2e/utils/test-utils.ts]: with `else`, code is clearer to me.

File: playwright-e2e/tests/dsm/participant/participant-sending-sample-received-creates-germline-consent-addendum.spec.ts [hunk 4]
```diff
@@ -470,7 +471,7 @@ test.describe.serial('Sending SAMPLE_RECEIVED event to DSS', () => {
  * Does not already have a germline consent addendum
  * Does not already have a normal kit (i.e. a blood or saliva kit) sent out (for ease of testing)
  */
-async function findParticipantForGermlineConsentCreation(participantListPage: ParticipantListPage, study: StudyEnum): Promise<string> {
+async function findParticipantForGermlineConsentCreation(page: Page, participantListPage: ParticipantListPage, study: StudyEnum): Promise<string> {
```
Comment (aweng98) [inline playwright-e2e/tests/dsm/participant/participant-sending-sample-received-creates-germline-consent-addendum.spec.ts]: new param `page: Page` is not needed because you can get the page object from `participantListPage` after changing access modifier of `page` variable to `public`.

**Review by aweng98 — APPROVED** [review]
left some comments. please make sure failed tests pass okay before merge. thank you.

File: playwright-e2e/dsm/pages/participant-list-page.ts [hunk 5]
```diff
@@ -245,10 +245,11 @@ export default class ParticipantListPage extends DsmPageBase {
     let firstName = '';
 
     const testParticipantFirstName = findPediatricParticipant ? user.child.firstName : user.adult.firstName; //Make sure to return automated test pts only
-    const responseJson = JSON.parse(await applyFilterResponse.text());
+    let responseJson = JSON.parse(await applyFilterResponse.text());
+    const amountOfParticipantsDisplayed = responseJson.participants.length;
 
     //Find a participant who currently has the specified tab
-    for (const index in responseJson.participants) {
+    for (let index = 0; index < amountOfParticipantsDisplayed; index++) {
```
Comment (aweng98) [inline playwright-e2e/dsm/pages/participant-list-page.ts]: `for` syntax minor update: 
`for (const [index, value] of [...responseJson.participants].entries())`

**Comment by Rinchan20** [comment]
running all tests passed in circleci in dev so merging on up!
